lib/src/builder_implementation.dart

```diff
@@ -2758,13 +2758,11 @@ class _ClassDomain {
   /// setters, and omits fields; in other words, it provides the
   /// behavioral point of view on the class. Also note that this is not
   /// the same semantics as that of `declarations` in [ClassMirror].
-  Iterable<ExecutableElement> get _declarations {
+  Iterable<ExecutableElement> get _declarations sync* {
     // TODO(sigurdm) feature: Include type variables (if we keep them).
-    return () sync* {
-      yield* _declaredMethods;
-      yield* _accessors;
-      yield* _constructors;
-    }();
+    yield* _declaredMethods;
```

Comment:
If it is not important that we construct this lazily we could just use:
```
[
  ..._declaredMethods,
  ..._acessors,
  ..._constructors,
]
```

Reply:
I think we're iterating over these collections in full and several times, so it's quite likely to be faster to compute lists. Done, including a few other `sync*` functions.